src/sentry/static/sentry/app/views/organizationIntegrations/integrationUtil.tsx

```diff
@@ -0,0 +1,37 @@
+const pluginMapping = {
+  id: 'id',
+  name: 'name',
+};
+
+const providerMapping = {
+  id: 'key',
+  name: 'name',
+};
+
+const sentryAppMapping = {
+  id: 'slug',
```

Comment:
@NisanthanNanthakumar We should use `slug` instead of `id` as the primary field we use here. When I think of `id`, I think of a DB incrementing `id`. Plugins have a `slug` field as well. This is especially confusing since SentryApps come from the DB and their `id` is an incrementing ID value.

Reply:
@scefali I see, but I think `key` would be better than `slug` to keep the same intent.

Reply:
@NisanthanNanthakumar I would highly prefer `slug` because it's the convention we follow with projects and organizations. We should be as consistent as possible with that terminology. I think there's a strong case for adding the `slug` to the integration provider so we can improve our consistency. @MeredithAnya What do you think?

Reply:
honestly I think the whole mapping here seems more overcomplicated than needed, since the `name` is always the same we can always do `integration.name` right? and then maybe the util function can just be `getIdentifier` or something so like if you could do 

```javascript
const slug = getIdentifier(integration, type)
```

```javascript
const getIdentifier= (integration, type) => {
  switch (type) {
    case 'plugin':
      return integration.id;
    case 'sentry-app':
      return integration.slug;
    case 'provider':
      return integration.key;
    default:
      return null;
  }
};
```
I'm a bit torn cause we use id in some places and slug in others, and then for the provider in this case we use key, but in general we use `slug` and the "id" for the plugin is more like a slug than the id so I would lean towards using that.

Reply:
Plugin has the `slug` so we can actually just do this:

```
const getIdentifier= (integration, type) => {
  switch (type) {
    case 'provider':
      return integration.key;
    default:
      return integration.slug;
  }
};
```

I think the mapping could be useful but I don't think this PR alone is demonstrating this.

Reply:
@scefali I'm okay with `slug`. @MeredithAnya For now, the mapping is just `slug` and `name`. But eventually, we could keep adding properties and use this to abstract over the 3 types of integrations. That's the goal.

Reply:
@NisanthanNanthakumar won't have to map the slug for long: https://github.com/getsentry/sentry/pull/16992

Long term, I would prefer to just change serializers to unify things instead of a bunch of conditional logic in our JS.